BREEZE.rst

```diff
@@ -46,409 +46,586 @@ Here is the short 10 minute video about Airflow Breeze
-See `Docker compose installation guide <https://docs.docker.com/compose/install/>`_
+There are three images that we are currently managing:
 
-Getopt and gstat
-----------------
+* **Slim CI** image that is used for static code checks (size of ~500MB). Its tag follows the pattern
+  of ``<BRANCH>-python<PYTHON_VERSION>-ci-slim`` (for example, ``apache/airflow:master-python3.6-ci-slim``).
+  The image is built using the `<Dockerfile>`_ Dockerfile.
+* **Full CI image*** that is used for testing. It contains a lot more test-related installed software
+  (size of ~1GB). Its tag follows the pattern of ``<BRANCH>-python<PYTHON_VERSION>-ci``
+  (for example, ``apache/airflow:master-python3.6-ci``). The image is built using the
+  `<Dockerfile>`_ Dockerfile.
+* **Checklicense image** that is used during license check with the Apache RAT tool. It does not
+  require any of the dependencies that the two CI images need so it is built using a different Dockerfile
+  `<Dockerfile-checklicence>`_ and only contains Java + Apache RAT tool. The image is
+  labelled with ``checklicence``, for example: ``apache/airflow:checklicence``. No versioning is used for
+  the Checklicence image.
 
-* If you are on MacOS
+Before you run tests, enter the environment or run local static checks, the necessary local images should be
+pulled and built from Docker Hub. This happens automatically for the test environment but you need to
+manually trigger it for static checks as described in `Building the images <#bulding-the-images>`_
+and `Pulling the latest images <#pulling-the-latest-images>`_.
+The static checks will fail and inform what to do if the image is not yet built.
 
-  * you need gnu ``getopt`` and ``gstat`` to get Airflow Breeze running.
+Building the image first time pulls a pre-built version of images from the Docker Hub, which may take some
+time. But for subsequent source code changes, no wait time is expected.
+However, changes to sensitive files like setup.py or Dockerfile will trigger a rebuild
+that may take more time though it is highly optimized to only rebuild what is needed.
 
-  * Typically you need to run ``brew install gnu-getopt coreutils`` and then follow instructions (you need to link the gnu getopt
-    version to become first on the PATH). Make sure to re-login after yoy make the suggested changes.
+In most cases, rebuilding an image requires network connectivity (for example, to download new
+dependencies). If you work offline and do not want to rebuild the images when needed, you can set the
+``FORCE_ANSWER_TO_QUESTIONS`` variable to ``no`` as described in the
+`Default behaviour for user interaction <#default-behaviour-for-user-interaction>`_ section.
 
-  * Then (with brew) link the gnu-getopt to become default as suggested by brew.
+See `Troubleshooting section <#troubleshooting>`_ for steps you can make to clean the environment.
 
-  * If you use bash, you should run this command (and re-login):
+Getopt and gstat
```

Comment:
```suggestion
Prerequisites 
```

Reply:
It's already a chapter in "Prerequisites" section :)